### PR TITLE
Apache Hive support

### PR DESCRIPTION
**Pull request by kaspersorensen**
This PR adds support for Apache Hive via the existing JDBC module.

To do this I had to introduce a new "isTransactional()" flag on IQueryRewriter since commit calls on Hive is not supported.

I also did some cleanup of Hadoop dependencies because I saw Hadoop version conflicts inbetween the "hadoop", "hbase" and now the "jdbc" modules. This caused the Hive connection to fail with class loading issues.

### CODE REVIEW

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java [hunk 1]
```diff
@@ -63,10 +63,12 @@ public void executePreparedStatement(PreparedStatement preparedStatement, boolea
...
         if (_connection == null) {
             _connection = getDataContext().getConnection();
-            try {
-                _connection.setAutoCommit(false);
-            } catch (SQLException e) {
-                throw JdbcUtils.wrapException(e, "disable auto-commit");
+            if (_dataContext.getQueryRewriter().isTransactional()) {
+                try {
+                    _connection.setAutoCommit(false);
+                } catch (SQLException e) {
```
Comment (albertostratio) [inline jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java]: With this change we are now setting the autocommit to false only when the isTransactional condition is true, is this what we want?

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java [hunk 2]
```diff
@@ -63,10 +63,12 @@ public void executePreparedStatement(PreparedStatement preparedStatement, boolea
     protected final Connection getConnection() {
         if (_connection == null) {
             _connection = getDataContext().getConnection();
-            try {
-                _connection.setAutoCommit(false);
-            } catch (SQLException e) {
-                throw JdbcUtils.wrapException(e, "disable auto-commit");
+            if (_dataContext.getQueryRewriter().isTransactional()) {
+                try {
```
Comment (albertostratio) [inline jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java]: With this change we are now setting the autocommit to false only when the isTransactional condition is true, is this what we want?

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java [hunk 3]
```diff
@@ -78,18 +80,20 @@ public final void close(boolean success) {
                 closePreparedStatement(_preparedStatement);
             }
 
-            try {
-                commitOrRollback(success);
+            if (_dataContext.getQueryRewriter().isTransactional()) {
+                try {
```
Comment (albertostratio) [inline jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java]: Same comment than above

**Comment by albertostratio** [comment]
@kaspersorensen I've added some comments, please review.

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java (hunk 2 above)
Comment (kaspersorensen) [inline jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java]: At least it's what I meant :-) We only have Hive in our portfolio currently that is a "non transactional" JDBC driver. In the case of Hive it is not allowed to call these methods.

Do you have an underlying concern with this specifically?

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java (hunk 3 above)
Comment (kaspersorensen) [inline jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java]: Same comment as above :)

File: jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java (hunk 2 above)
Comment (albertostratio) [inline jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java]: Ok got it! Thank you for clarifying, makes sense now.

**Comment by albertostratio** [comment]
Thank you @kaspersorensen 

These changes LGTM

**Pull request closed without merging**